api/v1alpha1/ext_auth_types.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	gwapiv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// ExtAuthServiceType specifies the types of External Authorization.
+// +kubebuilder:validation:Enum=GRPC;HTTP
+type ExtAuthServiceType string
+
+const (
+	// GRPC external authorization service.
+	GRPC ExtAuthServiceType = "GRPC"
+
+	// HTTP external authorization service.
+	HTTP ExtAuthServiceType = "HTTP"
+)
+
+// ExtAuth defines the configuration for External Authorization.
+type ExtAuth struct {
+	// Type decides the type of External Authorization.
+	// Valid ExtAuthServiceType values are "GRPC" or "HTTP".
+	//
+	// +unionDiscriminator
+	Type ExtAuthServiceType `json:"type"`
+
+	// GRPCService defines the gRPC External Authorization service
+	// Only one of GRPCService or HTTPService may be specified.
+	GRPCService *GRPCExtAuthService `json:"grpcService,omitempty" yaml:"grpcService"`
+
+	// HTTPService defines the HTTP External Authorization service
+	// Only one of GRPCService or HTTPService may be specified.
+	HTTPService *HTTPExtAuthService `json:"httpService,omitempty" yaml:"httpService"`
+
+	// AllowWhenFailed indicates whether the request should be allowed when the
+	// authorization service has failed or has returned an HTTP 5xx error.
+	// When set to true, the request will be allowed.
+	// When set to false, the request will be denied with an HTTP 403 Forbidden error.
+	// Defaults to false.
+	AllowWhenFailed bool `json:"allowWhenFailed,omitempty" yaml:"allowWhenFailed"`
```

Comment:
should be use `FailOpen` here ? also is this field needed for the first iteration of the API

Reply:
We can remove it from the initial API.